dbms/src/Flash/Pipeline/Schedule/TaskQueues/ResourceControlQueue.cpp

```diff
@@ -0,0 +1,223 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Flash/Executor/PipelineExecutorContext.h>
+#include <Flash/Executor/toRU.h>
+#include <Flash/Pipeline/Schedule/TaskQueues/IOPriorityQueue.h>
+#include <Flash/Pipeline/Schedule/TaskQueues/MultiLevelFeedbackQueue.h>
+#include <Flash/Pipeline/Schedule/TaskQueues/ResourceControlQueue.h>
+#include <Flash/Pipeline/Schedule/Tasks/TaskHelper.h>
+
+namespace DB
+{
+template <typename NestedTaskQueueType>
+void ResourceControlQueue<NestedTaskQueueType>::submit(TaskPtr && task)
+{
+    {
+        std::lock_guard lock(mu);
+        submitWithoutLock(std::move(task));
+    }
+    cv.notify_one();
+}
+
+template <typename NestedTaskQueueType>
+void ResourceControlQueue<NestedTaskQueueType>::submit(std::vector<TaskPtr> & tasks)
+{
+    {
+        std::lock_guard lock(mu);
+        for (auto & task : tasks)
+        {
+            submitWithoutLock(std::move(task));
+        }
+    }
+    cv.notify_one();
```

Comment:
Please call `notify_one` multiple times in `for` like `MLFQ`, or call `notify_all` here.
Because there may be multiple threads waiting at this time.

Reply:
done